tools/shared.py

```diff
@@ -396,6 +397,20 @@ def __init__(self):
       except Exception as e:
         exit_with_error(str(e) + 'Could not create canonical temp dir. Check definition of TEMP_DIR in ' + config.config_file_location())
 
+      # Since the cannonical temp directory is, by definition, that same
+      # between all processes that run in DEBUG mode we need to use a multi
+      # process lock to prevent more than on process from writing to it.
+      # This is because emcc assumes that it can use non-unique names inside
+      # the temp directory.
+      # In the case where we run emcc recurively to populate the cache we
+      # do still need to ignore this lock, in the same way we do for the cache
+      # lock.
+      if 'EM_EXCLUSIVE_CACHE_ACCESS' not in os.environ:
```

Comment:
it is a little unfortunate to mention `EM_EXCLUSIVE_CACHE_ACCESS` here as it would be the first time outside of `cache.py`. Perhaps we could call a helper function in that file?

Reply:
It is a little sad that need this exception here.. but I'm OK with hardcoding it for now.    Hopefully in the long run we can completely remove EM_EXCLUSIVE_CACHE_ACCESS.